packages/@vuepress/plugin-blog/index.js

```diff
@@ -65,7 +65,9 @@ module.exports = (options, ctx) => {
       }) => {
         if (when(pageCtx)) {
           Object.keys(frontmatter).forEach(key => {
-            rawFrontmatter[key] = rawFrontmatter[key] || frontmatter[key]
+            if (!pageCtx._hasInitLayout) {
+              rawFrontmatter[key] = frontmatter[key]
+            }
```

Comment:
Maybe it would block the apply of other attrs (e.g. `permalink `) in frontmatter ? actually I think the following override strategy should be right —— we should respect all the fields which has been set in user's frontmatter.

```js
if (!frontmatter[key]) {
  rawFrontmatter[key] || frontmatter[key]
}
```

Reply:
Yes, I ignored this question.

Reply:
https://github.com/vuejs/vuepress/blob/e01e60af2ddf475ff178fff2b674f2478f4b052f/packages/%40vuepress/plugin-blog/index.js#L68-L70

Is this setting a bit problematic?

Reply:
= instead of ||

Reply:
I corrected this by the way.

Reply:
cool